fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(res => {
+        return res.json()
+          .then(data => {
+            if (res.ok) return { todos: data, error: null };
+            return { todos: [], error: data };
+          })
+          .then(result => {
+            this.setState({ ...result });
+          })
+          .catch(err => console.error(err));
+      });
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+  }
+
+  addTodo(newTodo) {
+    const myHeaders = new Headers();
+    myHeaders.append('Content-Type', 'application/json');
+    const init = {
+      method: 'POST',
+      headers: myHeaders,
+      body: JSON.stringify(newTodo)
+    };
+    fetch('/api/todos', init)
+      .then(res => {
+        return res.json()
+          .then(data => {
+            if (res.ok) return { todos: this.state.todos.concat(newTodo), error: null };
+            return { todos: [...this.state.todos], error: data };
+          })
+          .catch(err => console.error(err));
+      });
+    this.componentDidMount();
```

Comment:
# 💀 
don't call `componentDidMount` or any other lifecycle method. those are for react only.